[POS Settings] Help section navigation

iamgabrielma opened the pull request:
Closes WOOMOB-1036

## Description
This PR adds the basic scaffolding and navigation for the Help section of POS settings, by allowing the merchant to navigate through help options and documentation in a detail view to the right, documentation and support are currently presented full screen (for now) on top of the view.

Final UI is not handled yet, this will be done separately.

https://indiemelon.mystagingwebsite.com/screen-recording-2025-08-26-at-11-17-01/

## Testing information
- In POS, tap on the `...` CTA, and go to Settings
- Tap Help
  - `Where are my products` will show the modal, and can redirect the merchant to order management if they wish to
  - `Documentation` will open a web view with the docs, full-screen
  - `Get support` opens the support form, once filled it's dismissed automatically and we're back to settings.

iamgabrielma committed:
render options for help section

iamgabrielma committed:
present supportForm

iamgabrielma committed:
present help and docs fullscreen

iamgabrielma committed:
localized strings

dangermattic commented:
<!--
  0 Errors
  1 Warning: View files have been modified,...
  0 Messages
  0 Markdowns
-->
<table>
  <thead>
    <tr>
      <th width="50"></th>
      <th width="100%" data-danger-table="true" data-kind="Warning">
          1 Warning
      </th>
     </tr>
  </thead>
  <tbody>
    <tr>
      <td>:warning:</td>
      <td data-sticky="false">View files have been modified, but no screenshot or video is included in the pull request. Consider adding some for clarity.</td>
    </tr>
  </tbody>
</table>

<p align="right" data-meta="generated_by_pr-check">
  Generated by :no_entry_sign: <a href="https://danger.systems/">Danger</a>
</p>

staskus commented on WooCommerce/Classes/POS/Presentation/Settings/PointOfSaleSettingsHelpDetailView.swift:
```diff
@@ -1,16 +1,151 @@
+            SafariView(url: WooConstants.URLs.pointOfSaleDocumentation.asURL())
+
+        }
+        .posFullScreenCover(isPresented: $showSupport) {
```
One thing that I haven't done and needs to be done, is scanning needs to be disabled when settings are opened. Maybe we could create a linear task for it or just do it now. Otherwise, filling out the support form in a simulator again triggers the scanning error sound. 

In `ItemListView` try adding `@EnvironmentObject var coverManager: POSFullScreenCoverManager` and in  `isBarcodeScanningEnabled` , add `&& !coverManager.isPresented`. Hopefully, that should be enough!

iamgabrielma replied:
Thanks for the suggestion! I tried it out and worked nicely. Updated: https://github.com/woocommerce/woocommerce-ios/pull/16039/commits/e2c260bcc4bc29e445168aac4ec648ba6730e29d

iamgabrielma committed:
disable scanner when POSFullScreenCoverManager is presented

Filling out the support form in the simulator will trigger the “not found” error sound from barcode scanning unless we disable it
